### DISCUSSION TITLE
Hi, i want to help with the german translation

### DISCUSSION THREAD

**Discussion by Stheel83** [sqlitebrowser/sqlitebrowser#3529] [General]
Hi I am Stephan, and I am located in Hamburg.

I want to join you to help with the german translation.
I work with DB Browser for SQlite since 2022 and now i want to give you something back.

In the past I did some localisation work für Opensuse with the tool "Lokalize".

But with the *.ts translation files, the program has some trouble right now, and i am not alone with it.
See this bugreport:
https://bugs.kde.org/show_bug.cgi?id=477779
https://www.mail-archive.com/kde-bugs-dist%40kde.org/msg863781.html
(it is from your polish translator: wojnilow)

Which program do you prefer? QT Linguist 5 / 6? Gitlocalize? Or something else?

############################

**Ok, now the main part.** 
I noticed some untranslated things during my work with DB Browser for SQlite. And therefore I did some analysis of the current (master) translation.

This is what i found till now and what i want to discuss. Afterwards then i want to change it, if possible:

_not or inconsistent translated:_
row(s) = Zeile / Zeilen (1x)
Constrain = Beschränkung (10x)
View = Ansichten (8x)
Toobar = Werkzeugleiste (8x)
Recently opened = zuletzt geöffnet (1x)
Recent Files = zuletzt geöffnete Dateien (1x)
Max Recent Files = Maximale Anzahl zuletzt geöffneter Dateien (1x)
Query = Abfrage (10x)
Create-Statement = (SQL-)Anweisung (3x)
SQL-Statement = SQL-Anweisung (1x)

############################

_short cuts:_
Ctrl = Strg (39x)
Shift = Umschalt (13x)
PgUp = BildAuf (1x)
PgDown = BildAb (1x)
Space = Leertaste (1x)
Del = Entf (2x)

############################

Rowid = ZeilenID (6x)

**moved this topic into a separate discussion:**
but, in the browse data Tab, when you activate "show rowid column", the column name is not translatable, it stays `_rowid_`
May it be possible to change this behavior?

############################

**moved this topic into a separate discussion:**
_not translatable in every language i tested:_ 
in "Edit Database Cell"open the context menu: 

- "Copy" and "Select All"

also in the "Execute SQL" tab open a SQL file. Then open the context menu inside the SQL Statement: 

- Undo, Redo, Cut, Copy, Paste, Delete, Select All

Looks like these are missing in the ExtendedScintilla.cpp, because "Find and Replace...", in the same context menu, is translatable.
But i can be completly wrong.

############################

_suggestion to change a sentence_
from this
() Die last_insert_rowid()-Funktion gibt die ROWID der letzte Zeile zurück, die von der diese Funktion aufrufenden Datenbankverbindung eingefügt wurde.
to this
() Die last_insert_rowid()-Funktion gibt die ZeilenID der letzten Zeile zurück, die von der Datenbankverbindung eingefügt wurde und die dann die Funktion aufrief.

############################

**moved this topic into a separate discussion:**
_Tab Edit Pragmas_
nearly the whole Tab is not translated, not in any language I tested and it looks like it is not translatable.
(Only the popup, when you hover over the check box from "Case Sensitive Like" is translatable)

I found hints to the Pragma-Tab in
MainWindow.cpp, MainWindow.h and

MainWindow.ui (please see Line 275)
`<string notr="true">&lt;html&gt;&lt;head/&gt;&lt;body&gt;&lt;p&gt;Checkpoint Full FSYNC &lt;a href=&quot;https://www.sqlite.org/pragma.html#pragma_checkpoint_fullfsync&quot;&gt;&lt;img src=&quot;:/icons/whatis&quot;/&gt;&lt;/a&gt;&lt;/a&gt;&lt;/p&gt;&lt;/body&gt;&lt;/html&gt;</string>`

I am not a programmer, but that looks not so easy to change.

############################

**moved this topic into a separate discussion:**
_Application Name_
what do you prefer: translate the App Name or let it stay in english?
I made a comparison und now i am unsure how you procede: 

- localised: de, pl, ar, cs, fr, id, nl, pt_BR, ru, sv, tr (still the old name "SQLite Browser"), uk_UA
- english: it, es, ja, ko_KR, zh, zh_TW

My oppinion: it is a name and it should stay in english. What do you think?

That are my points i want to discuss. 
I hope it is ok to put this all in one post for now and does not complicate to much.

Best regards
Stephan

**Comment by FriedrichFroebel** [comment]
The recommended tool for translations generally is Qt Linguist due to DB4S being written with the Qt framework itself as well: https://github.com/sqlitebrowser/sqlitebrowser/wiki/Translations

This being said, I recommend you to submit a corresponding PR with the easy fixes which allows for easy annotations/reviews of your proposed changes. I will happily have a look at it and I am open for discussions on some specific topics as well. While I tried my best when doing most of the current German translations and attempted to be consistent, some stuff might slip through - but DB4S is a FOSS project, thus everyone is free to further improve this.

For general stuff which might affect multiple languages, please open dedicated issues to allow for dedicated discussions/improvements. Handling multi-issue discussions as in your initial request generally is hard.

**Reply by Stheel83** [reply]
Hi Friedrich
many thanks for your reply and thank you for the link, that is very helpfull!
And thanks for your great work in the past and till now as a translator.
Ok, Qt linguist. I tested a little with it yesterday and it is not so easy to work with it. Maybe because i am used to Lokalize. I found the things i mentioned in Lokalize a lot easier than in QT Linguist.

As i am new to Github and this project: 
What do you mean with "corresponding PR"?
Can i enter my suggestions into the german *.ts file and post it here?
Sorry, but i don´t have a glue how to procede? ( In the past I only transmitted the translation per Mail to a Teammember)
Is there a wiki page i can read?

Ok, i will split the mulit language stuff into different posts.

**Reply by FriedrichFroebel** [reply]
> Ok, Qt linguist. I tested a little with it yesterday an it is not so easy to work with it. Maybe because i am used to Lokalize. I found the things i mentioned in Lokalize a lot easier than in QT Linguist.

This will always be a matter of taste of what works the best for you. Sometimes it just takes some time to get used to it and nobody forces you to use Qt Linguist - sometimes I just use a simple text editor as well, although it might lack some basic validation/guidance ;)

> What do you mean with "correspoinding PR"?
> Can i enter my suggestions into the german *.ts file and post it here?

I am referring to the default/usual GitHub workflow here, which roughly boils down to:

1. Create a fork of this repository.
2. On your own fork, create a new branch with a suitable name for your changes.
3. On your own fork, add your changes to the created branch and use a meaningful commit message to indicate what you have changed.
4. Go to this repository again. GitHub will usually propose you to generate a pull request with your own changes. Just click the button and fill out the template, then submit.

You will find quite some tutorials/howtos on the web or in the GitHub docs for specific aspects of these steps and for the general process to guide you through. Additionally, feel free to ask if you encounter a specific issue during the process.

**Reply by Stheel83** [reply]
Ok, thanks. 
I will have a look.

**Reply by Stheel83** [reply]
Ok
i did a pull request now. 
Lets see what you have to say and what i (maybe) have to change after all or what i forgot.

Just for info: I did the changes directly in the github editor now.

Thanks for now. 
Have a nice weekend.
